tests/backup/backup_helper.go

```diff
@@ -1223,11 +1404,64 @@ func restoreSuccessCheck(restoreName string, orgID string, retryDuration int, re
 
 	resp, err := backupDriver.InspectRestore(ctx, restoreInspectRequest)
 	if err != nil {
-		return false, err
+		return false, nil, err
 	}
 	restoreStatus := (resp.GetRestore().GetStatus().Status == api.RestoreInfo_StatusInfo_PartialSuccess) || (resp.GetRestore().GetStatus().Status == api.RestoreInfo_StatusInfo_Success)
 	log.Infof("[%s] restored successfully", restoreName)
-	return restoreStatus, nil
+	return restoreStatus, resp, nil
+}
+
+// returns the restoreContext
+func ValidateRestore(restoreInspectResponse *api.RestoreInspectResponse, backupContext *BackupRestoreContext, namespaceMapping map[string]string) (*BackupRestoreContext, error) {
+	//rstObjs := restoreInspectResponse.Restore.Resources
+	//TODO1 do the comparison of bkpObj and rstObj over here
+
+	rstCtx, err := GetRestoreCtxsFromBackupCtxs(*backupContext, namespaceMapping)
+	ValidateApplications(rstCtx.schedCtxs)
+
+	if err != nil {
+		return nil, fmt.Errorf("GetRestoreCtxsFromBackupCtxs Err: %v", err)
+	} else {
+		return &rstCtx, nil
+	}
+}
+
+// Call this function AFTER switching kubeconfig to the "restore" cluster
+func GetRestoreCtxsFromBackupCtxs(backupContext BackupRestoreContext, namespaceMapping map[string]string) (BackupRestoreContext, error) {
+	log.InfoD("Getting Restore Context from Backup Context")
+	var rstSchedCtxs []*scheduler.Context = make([]*scheduler.Context, 0)
+
+	options := CreateScheduleOptions()
+	for _, schedCtx := range backupContext.schedCtxs {
+		rstSchedCtx := *schedCtx
+		var specObjects []interface{}
+		specObjects = make([]interface{}, 0)
+
+		for _, appSpecOrig := range schedCtx.App.SpecList {
+			appSpec, err := CloneSpec(appSpecOrig) //clone spec to create "restore" specs
+			if err != nil {
+				log.Errorf("Failed to clone spec: '%v'. Err: %v", appSpecOrig, err)
```

Comment:
why you want to continue if there is an error

Reply:
That's because `CloneSpec` only errors out if the object which I'm trying to clone is unsupported by the function. This happens, for example, in case of CRs (like kind: elastic-search). This only affects the "cloning" of that particular object, and not the rest of the process. It can be safely ignored in most cases where cloning that particular object doesn't really matter, but if it does matter to someone, then the error will tell them that they need to care about it.

Technically speaking, the only case this occurs right now, is for CRs, for which I didn't write the necessary function as it's not important to "get things done", but it will cause an error (which can be ignored). I'll file bug for this and this can be removed when the bug is resolved. At the moment, this is definitely a P3 or P4 bug, so can be ignored.

can this be considered resolved for now?